packages/mongo-nav/src/org-nav/OrgNav.tsx

```diff
@@ -198,10 +219,12 @@ export default function OrgNav({
           onChange={onOrganizationChange}
           isActive={activeNav === 'orgSettings'}
           disabled={disabled}
+          isOnPrem={isOnPrem}
         />
         {!disabled && (
           <ul className={ulContainer}>
             {!isTablet &&
```

Comment:
This has gotten to a critical mass of conditionals where I think we may be better off moving the logic outside of the return statement. Something like this:

```ts
let badgeItem: React.ReactElement | null = null

if (
  !isTablet &&
  !isOnPrem &&
  current?.paymentStatus &&
  paymentVariant &&
  (admin || paymentValues.includes(current.paymentStatus))
) {
  badgeItem = (
    <li>
      <Badge
        className={css`margin-right: 25px;`}
        variant={paymentVariant}
        data-testid="org-payment-status"
      >
        {current.paymentStatus.split('_').join()}
      </Badge>
    </li>
  )
}

return (
  ...
  {badgeItem}
  ...
)
```

Reply:
Harry and I often disagree about when we should combine conditions, etc., and he usually comes down on the side of having more logic in the jsx is more readable, whereas I come down on the side of pulling out logic making things simpler for when you just need to grok code, etc.

That being said, I think nobody would disagree if we simplified any conditional rendering when there are three or more expressions.

Reply:
[aside] also, I wanted to share how we deal with responsive conditional rendering, etc. in cloud. We use `react-responsive`, which includes a hook `useMediaQuery` which handles detecting resizing, reading the window, etc., and takes in a min and max width as options.

We then apply that to create responsive wrapper components. For simplicity, I'm going to assign some arbitrary values and pull out types:

```
// breakpoint globals
const SMALL = 767;
const LARGE = 1025;
const MEDIUM_MIN = SMALL + 1;
const MEDIUM_MAX = LARGE - 1;

// internal helper
const getMediaQueryComponent = mediaQueryProps => {
  return ({ children }) => {
    const isMatchingSize = useMediaQuery(mediaQueryProps);
    return isMatchingSize ? children : none;
  };
};

// conditionally render content only for small viewports
const smallViewportProps = { maxWidth: SMALL };
export const SmallViewport = getMediaQueryComponent(smallViewportProps);

// conditionally render content for medium viewports ONLY
const mediumViewportProps = { minWidth: MEDIUM_MIN, maxWidth: MEDIUM_MAX };
export const MediumViewport = getMediaQueryComponent(mediumViewportProps);

// conditionally render content for medium viewports OR LARGER
const mediumPlusViewportProps = { minWidth: MEDIUM_MIN };
export const MediumPlusViewport = getMediaQueryComponent(mediumPlusViewportProps);
```

Which means we can. do things like
```
<SmallViewport>
  all my rendering for small screens can go here
</SmallViewport>
```
or, as needed, we can just use the hook mentioned at the top. 

**TL;DR I am totally not saying any of this should be changed**. I just wanted to expose the patterns we fell into for responsive design, in case they seem interesting or were different from what you found useful. We all don't do nearly as much responsive design as your average startup, etc, so I think it's good to share these things.

Reply:
👍